sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,902 @@
+                                 "the same classes as in the previous call to "
+                                 "fit. Previously got {}, `y` has {}".
+                                 format(self.classes_, classes))
+
+            if len(self.L_[0]) != len(X[0]):
+                raise ValueError('The new inputs dimensionality ({}) does not '
+                                 'match the previously learned transformation '
+                                 'input dimensionality ({}).'
+                                 .format(len(self.L_[0]), len(X[0])))
+        self.classes_ = classes
+
+        if self.n_features_out is not None:
+            check_scalar(self.n_features_out, 'n_features_out', int, 1)
+        check_scalar(self.n_neighbors, 'n_neighbors', int, 1, len(X) - 1)
+        check_scalar(self.max_iter, 'max_iter', int, 1)
+        check_scalar(self.max_constraints, 'max_constraints', int, 1)
+        check_scalar(self.max_corrections, 'max_corrections', int, 1)
+        check_scalar(self.n_jobs, 'n_jobs', int, -1)
+
+        check_scalar(self.tol, 'tol', float, 0.)
+
+        check_scalar(self.use_pca, 'use_pca', bool)
+        check_scalar(self.use_sparse, 'use_sparse', bool)
+        check_scalar(self.verbose, 'verbose', int, 0)
+
+        # Check linear transformation dimensions
+        if self.L is not None:
+            check_array(self.L)
+            if len(self.L[0]) != len(X[0]):
+                raise ValueError('Transformation input dimensionality ({}) '
+                                 'must match the inputs dimensionality ({}).'
+                                 .format(len(self.L[0]), len(X[0])))
+
+            if len(self.L) > len(self.L[0]):
+                raise ValueError('Transformation output dimensionality ({}) '
+                                 'cannot be greater than the '
+                                 'transformation input dimensionality ({}).'.
+                                 format(len(self.L), len(self.L[0])))
+
+        # Check preferred output dimensionality
+        if self.n_features_out is not None:
+            if self.L is not None:
+                if self.n_features_out != len(self.L):
+                    raise ValueError('Preferred outputs dimensionality ({}) '
+                                     'does not match the given linear '
+                                     'transformation {}!'.format(
+                                        self.n_features_out, len(self.L)))
+
+            elif self.n_features_out > len(X[0]):
+                raise ValueError('Preferred outputs dimensionality ({}) '
+                                 'cannot be greater than the given data '
+                                 'dimensionality {}!'.format(
+                                    self.n_features_out, len(X[0])))
+
+        # Check preferred number of neighbors
+        max_neighbors = min_class_size - 1
+        if self.n_neighbors > max_neighbors:
```

Comment:
Yeah, I think this would change other parts of the algorithm too, which rely on the target neighbor structure. Also, I think it's good to keep track as much with the original algorithm formulation.